main.go

```diff
@@ -176,3 +217,72 @@ func getModulePackageName(modFile *os.File, outputPath string) string {
 	moduleName = strings.TrimSpace(moduleName)
 	return fmt.Sprintf("%v/%v", moduleName, outputPath)
 }
+
+// validate validates the configuration and determines the goPackagePath for the generated code
+func (c *config) validate() error {
+	// check if the repo uses modules
+	if modFile, err := os.Open("go.mod"); err != nil {
+		if _, ok := err.(*os.PathError); !ok {
+			return fmt.Errorf("Error checking if the repo contains a go.mod file: %s", err.Error())
+		}
+		if *c.goPackageName == "" {
+			return fmt.Errorf("go-package is required")
+		}
+		c.goPackagePath = *c.goPackageName
+	} else {
+		defer modFile.Close()
+
+		goPath := os.Getenv("GOPATH")
+		if goPath == "" {
+			return fmt.Errorf("GOPATH must be set")
+		}
+		if *c.outputPath == "" {
+			return fmt.Errorf("output-path is required")
+		}
+
+		*c.outputPath = path.Clean(*c.outputPath)
+		c.goPackagePath = getModulePackagePath(goPath, *c.outputPath)
+		*c.goPackageName = getModulePackageName(modFile, *c.outputPath)
+	}
+
+	// parse flags determing if client/server code is generated
+	c.generateServer = true
+	c.generateGoClient = true
+	c.generateJSClient = true
+	// only generate client
+	if c.clientOnly != nil && *c.clientOnly == true {
```

Comment:
nit, we often use this style
```suggestion
	if swag.BoolValue(c.clientOnly) {
```